## 지도 학습 알고리즘 - 회귀 알고리즘의 사례

각 종양은 benign(양성)과 malignant(악성)으로 레이블링되어 있다. 전자는 해롭지 않은 종양, 후자는 암 종양에 해당한다.

In [28]:
import matplotlib.pyplot as plt
import mglearn
import numpy as np

In [21]:
from sklearn.datasets import load_breast_cancer
cancer = load_breast_cancer()
type(cancer)

sklearn.utils.Bunch

```Bunch``` 객체는 딕셔너리와 비슷하지만, ```cancer.data```처럼 점 표기법으로도 사용할 수 있다.

In [24]:
print("cancer.keys() :\n", cancer.keys())

cancer.keys() :
 dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename'])


In [34]:
print("유방암 데이터의 형태 :",cancer.data.shape)
print("클래스별 샘플 개수 :",
     {n : v for n, v in zip(cancer.target_names, np.bincount(cancer.target))})
print("특성 이름 :", cancer.feature_names)

유방암 데이터의 형태 : (569, 30)
클래스별 샘플 개수 : {'malignant': 212, 'benign': 357}
특성 이름 : ['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']


30개의 특성을 가지고 있는 569개의 데이터 포인트들로 이루어져 있다.